### ISSUE TITLE
CESM Workflows for running CESM model

### ISSUE THREAD

**Issue by annefou** [NordicESMhub/galaxy-tools#10]
- https://github.com/NCAR/CESM-WF

Creates a Cylc suite from the CESM XML environment.  Used to automate a CESM run.

These scripts are still in development - please contact Sheri Mickelson (mickelso@ucar.edu)
if you plan to use them in a production run.

**Comment by annefou** [comment]
With the latest version cesm 2.0.0, it should be available in the default distribution.

**Comment by annefou** [comment]
We have compiled and run CESM 2.1.0 (latest version of CESM) on ubuntu 18.04 and centos 7. 

CESM requires a recent GNU compiler (C, Fortran) and additional libraries (hdf5, netcdf, szip, mpich or openmpi) that need to be compiled with a recent compiler too. See https://github.com/NordicESMhub/containers/blob/master/cesm/2.1.0/README.md

**Comment by bgruening** [comment]
@annefou I created an initial conda recipes for CSEM, and I can compile it successfully I think. But I do not know what binaries are needed and should be moved into the package, maybe you can help?

I have this here:

```
insgesamt 2088
drwxr-xr-x  8 bag bag    4096 Apr 19 20:27 ./
drwxr-xr-x  8 bag bag    4096 Apr 19 20:36 ../
-rw-r--r--  1 bag bag    5123 Apr 19 20:26 build_env_setup.sh
-rw-rw-r--  1 bag bag 2060178 Dez  4 19:50 ChangeLog
-rw-rw-r--  1 bag bag    2433 Dez  4 19:50 ChangeLog_template
drwxr-xr-x 10 bag bag    4096 Apr 19 20:29 cime/
drwxrwxr-x  4 bag bag    4096 Dez  4 19:50 cime_config/
drwxr-xr-x 10 bag bag    4096 Apr 19 20:31 components/
-rwxrw-rw-  1 bag bag     259 Apr 19 20:26 conda_build.sh*
-rw-rw-r--  1 bag bag    3583 Dez  4 19:50 Copyright
drwxrwxr-x  3 bag bag    4096 Dez  4 19:50 doc/
-rw-rw-r--  1 bag bag    1441 Dez  4 19:50 Externals.cfg
drwxrwxr-x  2 bag bag    4096 Dez  4 19:50 .github/
-rw-rw-r--  1 bag bag     252 Dez  4 19:50 .gitignore
-rw-rw-r--  1 bag bag    2075 Dez  4 19:50 LICENSE.txt
drwxrwxr-x  5 bag bag    4096 Dez  4 19:50 manage_externals/
-rw-rw-r--  1 bag bag    8904 Dez  4 19:50 README.rst

```

**Comment by annefou** [comment]
We just discovered that we also need libnetcdf in conda-forge to be compiled in parallel. hdf5 is already available in conda-forge both as serial and parallel libraries so we should be able to follow the same "pattern" for libnetcdf (https://github.com/conda-forge/libnetcdf-feedstock). We have been discussing this issue at https://github.com/conda-forge/netcdf4-feedstock/issues/43

**Comment by annefou** [comment]
Under development. first prototype has been designed with planemo. 

Work to be done: for input data, we need to better organize (now just a tarball) so that users can make changes to test new hypothesis. We will first experiment using Galaxy Climate JupyterLab and then build the corresponding tool.

**Comment by annefou** [comment]
Done so closing this issue. Will open later a new issue with some improvement requested by users.
